tsconfig.json

```diff
@@ -13,7 +13,8 @@
     "resolveJsonModule": true,
     "declarationDir": "dist/types",
     "outDir": "dist/lib",
-    "typeRoots": ["node_modules/@types"]
+    "typeRoots": ["node_modules/@types"],
+    "downlevelIteration": true,
```

Comment:
We will have to check what impact this has on the gzip size. It would be unfortunate if we transpile into more expensive syntax for browsers that don't need it anymore. It would also be preferable if we had an explicit browser list.

Reply:
ESM doesn't change in size that much but UMD will gain `1kb`.

>  It would also be preferable if we had an explicit browser list.

If we still support IE11 the requirements can be very low as they are on es2015* which a lot of functions missing.,

Reply:
> It would be unfortunate if we transpile into more expensive syntax for browsers that don't need it anymore.

We could cheat and make `tslib` a required package so tslib will have the transpilation code and detect-gpu makes a simple call to that library to handle the iteration. Or just revert the code that needed this.

Reply:
> ESM doesn't change in size that much but UMD will gain `1kb`.

Woops - You meant gzip sizes. ESM doens't change at all, umd will gain 0.30kb

Reply:
About sizes speaking we can turn some switches on in terser to be a bit aggressive in optimizing.